### PR TITLE
Let payload provider render feature gate yaml

### PR DESCRIPTION
**Pull request by enxebre**
Currently we render the feature gate at the HCP level which results in a discrepancy when mco consume it along with a given NodePool payload-version that differs from the one used by the HCP to generate it. Related https://github.com/openshift/hypershift/pull/2643

CI Failure example
bootstrap.go:47] error running MCC[BOOTSTRAP]: error creating feature gate access: unable to determine features: missing desired version \"4.14.0-0.ci.test-2023-06-08-101141-ci-op-gvin17ww-initial\" in featuregates.config.openshift.io/cluster\n"}

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_machine-config-operator/3724/pull-ci-openshift-machine-config-operator-master-e2e-hypershift/1666749702499995648

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_machine-config-operator/3724/pull-ci-openshift-machine-config-operator-master-e2e-hypershift/1666749702499995648/artifacts/e2e-hypershift/run-e2e/artifacts/TestNodePool_PreTeardownClusterDump/namespaces/e2e-clusters-wvxmd-example-fmxk8/core/pods/logs/ignition-server-768699fd7b-w49lq-ignition-server.log

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/ignitionserver/ignitionserver.go [hunk 1]
```diff
@@ -437,7 +437,7 @@ func reconcileDeployment(deployment *appsv1.Deployment,
 						},
 						Args: []string{
 							"-c",
-							invokeFeatureGateRenderScript("/shared", releaseVersion, string(featureGateYAML)),
+							invokeFeatureGateRenderScript("/shared", string(featureGateYAML)),
```
Comment (csrwng) [inline control-plane-operator/controllers/hostedcontrolplane/ignitionserver/ignitionserver.go:440]: If all this container is doing is writing out the feature gate yaml, we can likely do without it. The local ignition provider could read the hcp directly and create the yaml.

**Review by csrwng — APPROVED** [review]
Just one comment, but if it fixes the current issue, we can definitely follow up.

**Comment by enxebre** [comment]
cluster config operator script failed with 
```
{"level":"info","ts":"2023-06-08T17:11:00Z","logger":"get-payload","msg":"cluster-config-operator process completed","time":"0s","output":"/bin/bash: line 19: /usr/bin/cluster-config-operator: No such file or directory\n"}
```
and

```
{"level":"info","ts":"2023-06-08T17:11:32Z","logger":"get-payload","msg":"cluster-config-operator process completed","time":"0s","output":"mkdir: cannot create directory 'input': File exists\nmkdir: cannot create directory 'output': File exists\nmkdir: cannot create directory 'manifests': File exists\n"}
```

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_hypershift/2664/pull-ci-openshift-hypershift-main-e2e-aws/1666849350090756096/artifacts/e2e-aws/run-e2e/artifacts/TestCreateCluster_PreTeardownClusterDump/namespaces/e2e-clusters-gtp7b-example-jqvp2/core/pods/logs/ignition-server-7b587d498c-vslgl-ignition-server.log
updated

**Comment by enxebre** [comment]
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_hypershift/2664/pull-ci-openshift-hypershift-main-e2e-aws/1667120840149307392/artifacts/e2e-aws/run-e2e/artifacts/TestNodePool_PreTeardownClusterDump/namespaces/e2e-clusters-55smd-example-498fs/core/pods/logs/ignition-server-7b9886b8b7-cf9d6-ignition-server.log
```
{"level":"info","ts":"2023-06-09T11:10:02Z","logger":"get-payload","msg":"cluster-config-operator process completed","time":"0s","output":"F0609 11:10:02.959705      33 render.go:53] failed rendering assets: lstat /usr/share/bootkube/manifests/bootstrap-manifests: no such file or directory\n"}
```

**Comment by enxebre** [comment]
we are back to only upgrade failing, now with
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_hypershift/2664/pull-ci-openshift-hypershift-main-e2e-aws/1668534680921575424/artifacts/e2e-aws/run-e2e/artifacts/TestNodePool_PreTeardownClusterDump/namespaces/e2e-clusters-kzpgc-example-sqbgk/core/pods/logs/ignition-server-567875cf5b-z8rws-ignition-server.log
```
{"level":"info","ts":"2023-06-13T08:52:01Z","logger":"get-payload","msg":"cluster-config-operator process completed","time":"0s","output":"Error: unknown flag: --rendered-manifest-files\n"}
```

**Comment by JoelSpeed** [comment]
> we are back to only upgrade failing, now with https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_hypershift/2664/pull-ci-openshift-hypershift-main-e2e-aws/1668534680921575424/artifacts/e2e-aws/run-e2e/artifacts/TestNodePool_PreTeardownClusterDump/namespaces/e2e-clusters-kzpgc-example-sqbgk/core/pods/logs/ignition-server-567875cf5b-z8rws-ignition-server.log
> 
> ```
> {"level":"info","ts":"2023-06-13T08:52:01Z","logger":"get-payload","msg":"cluster-config-operator process completed","time":"0s","output":"Error: unknown flag: --rendered-manifest-files\n"}
> ```

This looks like a mismatch between the payload version and the flags, is the test upgrading from a 4.14 to 4.14 or from 4.13 to 4.14? If it is upgrading from 4.13, we need to make sure it only passes the `--rendered-manifest-files` flag to the 4.14 payload not the 4.13 payload

File: ignition-server/controllers/local_ignitionprovider.go [hunk 2]
```diff
@@ -678,3 +708,50 @@ func findMatchingManifest(imageRef string, deserializedManifestList *manifestlis
...
+
+touch %[2]s/manifests/99_feature-gate.yaml
+cat <<EOF >%[2]s/manifests/99_feature-gate.yaml
+%[5]s
+EOF
+
+%[1]s render \
+   --config-output-file config \
+   --asset-input-dir %[2]s/input \
+   --asset-output-dir %[2]s/output
```
[the remark this answers is not in the document]
Comment (enxebre) [reply ignition-server/controllers/local_ignitionprovider.go:751]: no, only for >= 4.14

**Comment by enxebre** [comment]
>{Failed  === RUN   TestUpgradeControlPlane/EnsureNoCrashingPods
    util.go:457: Container oauth-server in pod oauth-openshift-74bf787c59-6gh9p has a restartCount > 0 (2)

/test e2e-aws

**Pull request merged**